experimental/batchmap/tilepb/tile.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package tilepb;
+
+option go_package = "tilepb";
+
+// Tile represents a perfect subtree covering the whole height of a stratum
+// within a sparse map.
+message Tile {
+  // The path from the root of the map to the root of this tile.
+  bytes path = 1;
+  // The computed hash of this subtree.
+  // TODO(mhutchinson): Consider removing this.
+  bytes root_hash = 2;
+  // All non-empty leaves in this tile, sorted left-to-right.
+  repeated TileLeaf leaves = 3;
+}
+
+// TileLeaf is a leaf value of a Tile.
+// If it belongs to a leaf tile then this represents one of the values that the
+// map commits to. Otherwise, this leaf represents the root of the subtree in
+// the stratum below.
+message TileLeaf {
+  // The path from the root of the container Tile to this leaf.
+  bytes path = 1;
+  // The hash value being committed to.
+  bytes hash = 2;
+}
+
+// Entry is a single key/value to be committed to by the map.
+message Entry {
+  // The key that uniquely identifies this key/value.
+  // These keys must be distributed uniformly and randomly across the key space.
```

Comment:
Maybe worth just connecting up the dots and explicitly saying we use a cryptographically secure hash of the application "key" to provide this property?

Reply:
"We" don't do this, where "we" are the library maintainers. The `Entry` proto is provided as input to the map by the client. But I've extended this to say it should be obtained using a crypto hash.